File: php/power.php

<?php

// given a base and exponent, calculate the power recursively

$base = $argv[1];
$exponent = $argv[2];

function power($bs, $exp) {
    if($exp === 0) {
        return 1;
    } else {
        return $bs * power($bs, $exp - 1);
    }
};

echo power($base, $exponent) . "\n";

?>

File: ruby/power.rb

# given a base and exponent, calculate the power recursively

base = ARGV[0].to_f
exponent = ARGV[1].to_f

def power(b, e)
    if e == 0
        return 1
    else
        return b * power(b, e - 1)
    end
end

result = power(base, exponent)

puts result % 1 == 0 ? result.to_i : result
